From 2a5081bfdfd60d3b7b4415e552354ac8e711a9d2 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Tue, 7 Nov 2023 12:51:01 +0530
Subject: [PATCH 135/135] Enabled Dbus Sel Logging Support

Description:
     Enabled Dbus Sel Logging Support and Added Proper Success Message.

Test case:
root@AMIOT-9e6399525847:~# busctl tree xyz.openbmc_project.Logging
`-/xyz
  `-/xyz/openbmc_project
    `-/xyz/openbmc_project/logging
      |-/xyz/openbmc_project/logging/entry
      | |-/xyz/openbmc_project/logging/entry/1
      | |-/xyz/openbmc_project/logging/entry/2
      | |-/xyz/openbmc_project/logging/entry/3
      | |-/xyz/openbmc_project/logging/entry/4
      | |-/xyz/openbmc_project/logging/entry/5
      `-/xyz/openbmc_project/logging/internal
        `-/xyz/openbmc_project/logging/internal/manager
1) Get EventLog 
Get Call
https://x.x.x.x/redfish/v1/Systems/system/LogServices/EventLog

Response:
{
    "@odata.id": "/redfish/v1/Systems/system/LogServices/EventLog",
    "@odata.type": "#LogService.v1_1_0.LogService",
    "Actions": {
        "#LogService.ClearLog": {
            "target": "/redfish/v1/Systems/system/LogServices/EventLog/Actions/LogService.ClearLog"
        }
    },
    "DateTime": "2023-11-07T06:31:01+00:00",
    "DateTimeLocalOffset": "+00:00",
    "Description": "System Event Log Service",
    "Entries": {
        "@odata.id": "/redfish/v1/Systems/system/LogServices/EventLog/Entries"
    },
    "Id": "EventLog",
    "Name": "Event Log Service",
    "OverWritePolicy": "WrapsWhenFull"
}
2)Get EventLog Entries
Get Call
https://x.x.x.x/redfish/v1/Systems/system/LogServices/EventLog/Entries/{{Entries_Instance}} 

Response:
{
    "@odata.id": "/redfish/v1/Systems/system/LogServices/EventLog/Entries/1",
    "@odata.type": "#LogEntry.v1_9_0.LogEntry",
    "AdditionalDataURI": "/redfish/v1/Systems/system/LogServices/EventLog/Entries/1/attachment",
    "Created": "2023-11-07T06:29:35.536+00:00",
    "EntryType": "Event",
    "Id": "1",
    "Message": "P105_PCH_AUX critical high threshold assert. Reading=1.344300 Threshold=1.139000.",
    "Modified": "2023-11-07T06:29:35.536+00:00",
    "Name": "System Event Log Entry",
    "Resolved": false,
    "Severity": "OK"
}

3)Update Event log Status
Patch Call
https://x.x.x.x/redfish/v1/Systems/system/LogServices/EventLog/Entries/1

Body:
{
    "Resolved":true
}

Response:
{
    "@Message.ExtendedInfo": [
        {
            "@odata.type": "#Message.v1_1_1.Message",
            "Message": "The request completed successfully.",
            "MessageArgs": [],
            "MessageId": "Base.1.16.0.Success",
            "MessageSeverity": "OK",
            "Resolution": "None"
        }
    ]
}

4)Delete Event Log 

Delete Call
https://x.x.x.x/redfish/v1/Systems/system/LogServices/EventLog/Entries/1

Response:
{
    "@Message.ExtendedInfo": [
        {
            "@odata.type": "#Message.v1_1_1.Message",
            "Message": "The request completed successfully.",
            "MessageArgs": [],
            "MessageId": "Base.1.16.0.Success",
            "MessageSeverity": "OK",
            "Resolution": "None"
        }
    ]
}

5)Delete All EventLog

Post call
https://x.x.x.x/redfish/v1/Systems/system/LogServices/EventLog/Actions/LogService.ClearLog

Response:
{
    "@Message.ExtendedInfo": [
        {
            "@odata.type": "#Message.v1_1_1.Message",
            "Message": "The request completed successfully.",
            "MessageArgs": [],
            "MessageId": "Base.1.16.0.Success",
            "MessageSeverity": "OK",
            "Resolution": "None"
        }
    ]
}
---
 redfish-core/lib/event_service.hpp | 2 +-
 redfish-core/lib/log_services.hpp  | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/redfish-core/lib/event_service.hpp b/redfish-core/lib/event_service.hpp
index e479386..94ee99a 100644
--- a/redfish-core/lib/event_service.hpp
+++ b/redfish-core/lib/event_service.hpp
@@ -264,7 +264,7 @@ inline void requestRoutesSubmitTestEvent(App& app)
         {
             return;
         }
-        EventServiceManager::getInstance().readEventLogsFromFile();
+        // EventServiceManager::getInstance().readEventLogsFromFile();
         asyncResp->res.result(boost::beast::http::status::no_content);
     });
 }
diff --git a/redfish-core/lib/log_services.hpp b/redfish-core/lib/log_services.hpp
index 0ec639e..232be45 100644
--- a/redfish-core/lib/log_services.hpp
+++ b/redfish-core/lib/log_services.hpp
@@ -1902,6 +1902,7 @@ inline void requestRoutesDBusEventLogEntry(App& app)
                 messages::internalError(asyncResp->res);
                 return;
             }
+            messages::success(asyncResp->res);
         });
     });
 
@@ -1957,7 +1958,8 @@ inline void requestRoutesDBusEventLogEntry(App& app)
                 return;
             }
 
-            asyncResp->res.result(boost::beast::http::status::ok);
+            // asyncResp->res.result(boost::beast::http::status::ok);
+            messages::success(asyncResp->res);
         };
 
         // Make call to Logging service to request Delete Log
@@ -3679,7 +3681,7 @@ inline void requestRoutesDBusLogServiceActionsClear(App& app)
                 return;
             }
 
-            asyncResp->res.result(boost::beast::http::status::no_content);
+            messages::success(asyncResp->res);
         };
 
         // Make call to Logging service to request Clear Log
-- 
2.9.5

